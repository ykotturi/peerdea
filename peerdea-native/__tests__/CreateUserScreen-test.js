import React from 'react';
import renderer from 'react-test-renderer';

import CreateUserScreen from '../screens/CreateUserScreen';

jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);
jest.mock('react-native-keyboard-aware-scroll-view', () => {
  return {
    KeyboardAwareScrollView: jest
      .fn()
      .mockImplementation(({ children }) => children),
  };
});

describe('Create User Screen', () => {
  jest.useFakeTimers();

  it(`renders the create user screen`, () => {
    const tree = renderer.create(<CreateUserScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
