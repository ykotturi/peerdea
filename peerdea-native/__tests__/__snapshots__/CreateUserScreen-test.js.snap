// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create User Screen renders the create user screen 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "flex": 1,
      "paddingBottom": 30,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "backgroundColor": "#fff",
        "width": "100%",
      }
    }
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "flex": 1,
              "justifyContent": "center",
              "paddingBottom": 30,
              "paddingTop": 18,
            }
          }
        >
          <Image
            accessibilityLabel="Peerdea app logo"
            accessible={true}
            source={1}
            style={
              Object {
                "height": 51,
                "marginBottom": 10,
                "width": 300,
              }
            }
          />
          <Text
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Upload or take a profile picture
          </Text>
          <View
            style={
              Array [
                Object {
                  "padding": 0,
                },
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "https://t3.ftcdn.net/jpg/02/68/55/60/240_F_268556012_c1WBaKFN5rjRxR2eyV33znK4qnYeKZjm.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 30,
                  "height": 637.5,
                  "opacity": 0.6,
                  "width": 637.5,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-around",
                "marginBottom": 10,
                "width": "95%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0076ff",
                  "borderColor": "#0076ff",
                  "borderRadius": 9,
                  "borderWidth": 1,
                  "elevation": 3,
                  "height": 50,
                  "justifyContent": "center",
                  "width": "47%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignItems": "center",
                    "color": "#ffffff",
                    "fontSize": 20,
                    "justifyContent": "center",
                    "paddingHorizontal": "8%",
                  }
                }
              >
                Camera
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0076ff",
                  "borderColor": "#0076ff",
                  "borderRadius": 9,
                  "borderWidth": 1,
                  "elevation": 3,
                  "height": 50,
                  "justifyContent": "center",
                  "width": "47%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignItems": "center",
                    "color": "#ffffff",
                    "fontSize": 20,
                    "justifyContent": "center",
                    "paddingHorizontal": "8%",
                  }
                }
              >
                Camera Roll
              </Text>
            </View>
          </View>
          <TextInput
            blurOnSubmit={false}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Enter a username (required)"
            placeholderTextColor="grey"
            returnKeyType="next"
            style={
              Object {
                "borderColor": "gray",
                "borderRadius": 5,
                "borderWidth": 1,
                "height": 40,
                "marginBottom": 10,
                "paddingHorizontal": 3,
                "width": "80%",
              }
            }
            textContentType="oneTimeCode"
          />
          <TextInput
            blurOnSubmit={false}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Enter a password (required)"
            placeholderTextColor="grey"
            returnKeyType="next"
            secureTextEntry={true}
            style={
              Object {
                "borderColor": "gray",
                "borderRadius": 5,
                "borderWidth": 1,
                "height": 40,
                "marginBottom": 10,
                "paddingHorizontal": 3,
                "width": "80%",
              }
            }
            textContentType="oneTimeCode"
          />
          <TextInput
            blurOnSubmit={false}
            keyboardType="email-address"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Enter your email (required)"
            placeholderTextColor="grey"
            returnKeyType="next"
            style={
              Object {
                "borderColor": "gray",
                "borderRadius": 5,
                "borderWidth": 1,
                "height": 40,
                "marginBottom": 10,
                "paddingHorizontal": 3,
                "width": "80%",
              }
            }
          />
          <TextInput
            blurOnSubmit={false}
            multiline={true}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Enter a brief bio describing yourself or your work! (required)"
            placeholderTextColor="grey"
            returnKeyType="next"
            style={
              Object {
                "borderColor": "gray",
                "borderRadius": 5,
                "borderWidth": 1,
                "height": 40,
                "marginBottom": 10,
                "paddingHorizontal": 3,
                "width": "80%",
              }
            }
          />
          <TextInput
            blurOnSubmit={false}
            multiline={true}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Enter a 6 month goal for your business! (required)"
            placeholderTextColor="grey"
            returnKeyType="done"
            style={
              Object {
                "borderColor": "gray",
                "borderRadius": 5,
                "borderWidth": 1,
                "height": 40,
                "marginBottom": 10,
                "paddingHorizontal": 3,
                "width": "80%",
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#0076ff",
                "borderColor": "#0076ff",
                "borderRadius": 9,
                "borderWidth": 1,
                "elevation": 3,
                "height": 50,
                "justifyContent": "center",
                "width": "55%",
              }
            }
          >
            <Text
              style={
                Object {
                  "alignItems": "center",
                  "color": "#ffffff",
                  "fontSize": 20,
                  "justifyContent": "center",
                  "paddingHorizontal": "8%",
                }
              }
            >
              Sign up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  />
</View>
`;
